Adopt more smart pointers under WebCore/page/

cdumez opened the pull request:
#### cac77a010b989177ff66d391dacea5563cd4f48f
<pre>
Adopt more smart pointers under WebCore/page/
<a href="https://bugs.webkit.org/show_bug.cgi?id=269209">https://bugs.webkit.org/show_bug.cgi?id=269209</a>

Reviewed by Brent Fulgham.

* Source/WebCore/bindings/js/JSWindowProxy.cpp:
(WebCore::JSWindowProxy::protectedWorld):
* Source/WebCore/bindings/js/JSWindowProxy.h:
* Source/WebCore/dom/Node.cpp:
(WebCore::Node::protectedContainingShadowRoot const):
(WebCore::Node::checkedRenderer const):
* Source/WebCore/dom/Node.h:
* Source/WebCore/html/HTMLFrameOwnerElement.cpp:
(WebCore::HTMLFrameOwnerElement::~HTMLFrameOwnerElement):
(WebCore::HTMLFrameOwnerElement::protectedContentFrame const):
* Source/WebCore/html/HTMLFrameOwnerElement.h:
* Source/WebCore/inspector/agents/page/PageRuntimeAgent.cpp:
(WebCore::PageRuntimeAgent::PageRuntimeAgent):
(WebCore::PageRuntimeAgent::reportExecutionContextCreation):
* Source/WebCore/page/DOMSelection.cpp:
(WebCore::DOMSelection::getComposedRanges):
* Source/WebCore/page/EventHandler.cpp:
(WebCore::EventHandler::updateSelectionForMouseDrag):
(WebCore::EventHandler::hitTestResultAtPoint const):
(WebCore::EventHandler::performDragAndDrop):
(WebCore::EventHandler::dispatchMouseEvent):
(WebCore::EventHandler::sendContextMenuEvent):
(WebCore::EventHandler::dispatchEventToDragSourceElement):
(WebCore::EventHandler::handleDrag):
(WebCore::EventHandler::keyboardScrollRecursively):
* Source/WebCore/page/FocusController.cpp:
(WebCore::FocusController::setFocusedFrame):
(WebCore::FocusController::focusedOrMainFrame const):
(WebCore::FocusController::setFocused):
(WebCore::FocusController::setFocusedInternal):
(WebCore::FocusController::relinquishFocusToChrome):
(WebCore::FocusController::advanceFocusInDocumentOrder):
(WebCore::shouldClearSelectionWhenChangingFocusedElement):
(WebCore::FocusController::setFocusedElement):
(WebCore::FocusController::protectedPage const):
(WebCore::FocusController::setActive):
(WebCore::FocusController::setActiveInternal):
(WebCore::FocusController::setIsVisibleAndActiveInternal):
(WebCore::FocusController::advanceFocusDirectionallyInContainer):
* Source/WebCore/page/FocusController.h:
* Source/WebCore/page/LocalDOMWindow.cpp:
(WebCore::LocalDOMWindow::shouldHaveWebKitNamespaceForWorld):
(WebCore::LocalDOMWindow::webkitNamespace):
(WebCore::LocalDOMWindow::isAllowedToUseDeviceMotion const):
(WebCore::LocalDOMWindow::open):
* Source/WebCore/page/LocalFrame.cpp:
(WebCore::LocalFrame::screenSize const):
* Source/WebCore/page/LocalFrameView.cpp:
(WebCore::LocalFrameView::updateSnapOffsets):
* Source/WebCore/page/LocalFrameViewLayoutContext.cpp:
(WebCore::LocalFrameViewLayoutContext::layout):
(WebCore::LocalFrameViewLayoutContext::performLayout):
(WebCore::LocalFrameViewLayoutContext::runPostLayoutTasks):
(WebCore::LocalFrameViewLayoutContext::scheduleLayout):
(WebCore::LocalFrameViewLayoutContext::scheduleSubtreeLayout):
(WebCore::LocalFrameViewLayoutContext::updateStyleForLayout):
(WebCore::LocalFrameViewLayoutContext::protectedFrame):
(WebCore::LocalFrameViewLayoutContext::view const):
(WebCore::LocalFrameViewLayoutContext::protectedView const):
* Source/WebCore/page/LocalFrameViewLayoutContext.h:
* Source/WebCore/page/Location.cpp:
(WebCore::Location::reload):
* Source/WebCore/page/MouseEventWithHitTestResults.h:
(WebCore::MouseEventWithHitTestResults::protectedTargetNode const):
* Source/WebCore/page/Navigator.cpp:
(WebCore::Navigator::appVersion const):
(WebCore::Navigator::userAgent const):
(WebCore::Navigator::plugins):
(WebCore::Navigator::mimeTypes):
(WebCore::Navigator::cookieEnabled const):
* Source/WebCore/page/Page.cpp:
(WebCore::Page::viewportArguments const):
(WebCore::Page::takeAnyMediaCanStartListener):
(WebCore::Page::setCanStartMedia):
(WebCore::Page::protectedMainFrame const):
(WebCore::Page::forEachDocument const):
* Source/WebCore/page/Page.h:
* Source/WebCore/page/PageConsoleClient.cpp:
(WebCore::PageConsoleClient::addMessage):
(WebCore::PageConsoleClient::messageWithTypeAndLevel):
(WebCore::PageConsoleClient::count):
(WebCore::PageConsoleClient::countReset):
(WebCore::PageConsoleClient::profile):
(WebCore::PageConsoleClient::profileEnd):
(WebCore::PageConsoleClient::takeHeapSnapshot):
(WebCore::PageConsoleClient::time):
(WebCore::PageConsoleClient::timeLog):
(WebCore::PageConsoleClient::timeEnd):
(WebCore::PageConsoleClient::timeStamp):
(WebCore::PageConsoleClient::screenshot):
(WebCore::PageConsoleClient::protectedPage const):
* Source/WebCore/page/PageConsoleClient.h:
* Source/WebCore/page/PageOverlayController.cpp:
(WebCore::PageOverlayController::createRootLayersIfNeeded):
(WebCore::PageOverlayController::installedPageOverlaysChanged):
(WebCore::PageOverlayController::protectedPage const):
(WebCore::PageOverlayController::attachViewOverlayLayers):
(WebCore::PageOverlayController::detachViewOverlayLayers):
(WebCore::PageOverlayController::layerWithDocumentOverlays):
(WebCore::PageOverlayController::layerWithViewOverlays):
(WebCore::PageOverlayController::protectedLayerWithViewOverlays):
(WebCore::PageOverlayController::installPageOverlay):
(WebCore::PageOverlayController::updateForceSynchronousScrollLayerPositionUpdates):
(WebCore::PageOverlayController::didChangeViewExposedRect):
(WebCore::PageOverlayController::updateSettingsForLayer):
(WebCore::PageOverlayController::deviceScaleFactor const):
(WebCore::PageOverlayController::notifyFlushRequired):
(WebCore::PageOverlayController::didChangeOverlayFrame):
(WebCore::PageOverlayController::didChangeOverlayBackgroundColor):
(WebCore::PageOverlayController::tiledBackingUsageChanged):
* Source/WebCore/page/PageOverlayController.h:
* Source/WebCore/page/PerformanceMonitor.cpp:
(WebCore::PerformanceMonitor::didFinishLoad):
(WebCore::PerformanceMonitor::activityStateChanged):
(WebCore::PerformanceMonitor::measurePostLoadCPUUsage):
(WebCore::PerformanceMonitor::measurePostLoadMemoryUsage):
(WebCore::PerformanceMonitor::measurePostBackgroundingMemoryUsage):
(WebCore::PerformanceMonitor::measurePostBackgroundingCPUUsage):
(WebCore::PerformanceMonitor::measurePerActivityStateCPUUsage):
(WebCore::PerformanceMonitor::measureCPUUsageInActivityState):
* Source/WebCore/page/PerformanceMonitor.h:
* Source/WebCore/page/PointerLockController.cpp:
(WebCore::PointerLockController::didAcquirePointerLock):
(WebCore::PointerLockController::didNotAcquirePointerLock):
(WebCore::PointerLockController::didLosePointerLock):
(WebCore::PointerLockController::enqueueEvent):
* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::needsAutoplayPlayPauseEvents const):
(WebCore::Quirks::needsPerDocumentAutoplayBehavior const):
(WebCore::Quirks::shouldSynthesizeTouchEvents const):
(WebCore::Quirks::shouldBypassBackForwardCache const):
(WebCore::Quirks::shouldAvoidPastingImagesAsWebContent const):
(WebCore::Quirks::requestStorageAccessAndHandleClick const):
(WebCore::Quirks::protectedDocument const):
(WebCore::Quirks::triggerOptionalStorageAccessIframeQuirk const):
(WebCore::Quirks::triggerOptionalStorageAccessQuirk const):
(WebCore::Quirks::needsDisableDOMPasteAccessQuirk const):
* Source/WebCore/page/Quirks.h:
* Source/WebCore/page/ResourceUsageOverlay.cpp:
(WebCore::ResourceUsageOverlay::~ResourceUsageOverlay):
(WebCore::ResourceUsageOverlay::initialize):
(WebCore::ResourceUsageOverlay::mouseEvent):
* Source/WebCore/page/Screen.cpp:
(WebCore::fingerprintingProtectionsEnabled):
(WebCore::Screen::height const):
(WebCore::Screen::width const):
(WebCore::Screen::colorDepth const):
(WebCore::Screen::availLeft const):
(WebCore::Screen::availTop const):
(WebCore::Screen::availHeight const):
(WebCore::Screen::availWidth const):
(WebCore::Screen::orientation):
* Source/WebCore/page/SpatialNavigation.cpp:
(WebCore::FocusCandidate::FocusCandidate):
(WebCore::canScrollInDirection):
(WebCore::nodeRectInAbsoluteCoordinates):
(WebCore::virtualRectForAreaElementAndDirection):
(WebCore::frameOwnerElement):
* Source/WebCore/page/SpatialNavigation.h:
* Source/WebCore/page/WebKitNamespace.cpp:
(WebCore::WebKitNamespace::WebKitNamespace):
* Source/WebCore/page/cocoa/WebTextIndicatorLayer.mm:
(-[WebTextIndicatorLayer present]):
* Source/WebCore/page/mac/EventHandlerMac.mm:
(WebCore::EventHandler::passWidgetMouseDownEventToWidget):
* Source/WebCore/page/mac/ServicesOverlayController.h:
* Source/WebCore/page/mac/ServicesOverlayController.mm:
(WebCore::ServicesOverlayController::invalidateHighlightsOfType):
(WebCore::ServicesOverlayController::buildPotentialHighlightsIfNeeded):
(WebCore::ServicesOverlayController::remainingTimeUntilHighlightShouldBeShown const):
(WebCore::ServicesOverlayController::buildPhoneNumberHighlights):
(WebCore::ServicesOverlayController::buildSelectionHighlight):
(WebCore::ServicesOverlayController::hasRelevantSelectionServices):
(WebCore::ServicesOverlayController::createOverlayIfNeeded):
(WebCore::ServicesOverlayController::telephoneNumberRangesForFocusedFrame):
(WebCore::ServicesOverlayController::findTelephoneNumberHighlightContainingSelectionHighlight):
(WebCore::ServicesOverlayController::mouseEvent):
(WebCore::ServicesOverlayController::handleClick):
* Source/WebCore/rendering/HitTestResult.cpp:
(WebCore::HitTestResult::protectedTargetNode const):
(WebCore::HitTestResult::protectedTargetElement const):
* Source/WebCore/rendering/HitTestResult.h:
* Source/WebCore/rendering/RenderWidget.h:
(WebCore::RenderWidget::protectedWidget const):

Canonical link: <a href="https://commits.webkit.org/274488@main">https://commits.webkit.org/274488@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/348f1efc647dadc7ed5887e6486afc7805b62d62

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/39225 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/18204 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/41578 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/41759 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/35125 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/41531 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/21057 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/15533 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/32829 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/39799 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/15322 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/34002 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/13313 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/13282 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/34941 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/43037 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/35617 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/35271 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/39104 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/14037 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/11591 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/37342 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/15643 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/15306 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/5127 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/15129 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebCore/page/FocusController.cpp:
```diff
@@ -925,32 +926,33 @@ bool FocusController::setFocusedElement(Element* element, LocalFrame& newFocused
         if (oldDocument)
             oldDocument->setFocusedElement(nullptr);
-        m_page.editorClient().setInputMethodState(nullptr);
+        page->editorClient().setInputMethodState(nullptr);
```
Do we need a `protectedEditorClient`?

cdumez replied:
EditorClient is not refcounted. Protecting the Page is sufficient to keep it alive. It should ideally be a `checkedEditorClient()` call but it doesn't exist yet and I was focusing on issues actually reported by the static analyzer to make faster progress.

brentfulgham commented on Source/WebCore/page/LocalFrameViewLayoutContext.cpp:
```diff
@@ -528,22 +529,22 @@ void LocalFrameViewLayoutContext::applyTextSizingIfNeeded(RenderElement& layoutR
     // FIXME: This shouldn't be necessary, but see rdar://problem/36670246.
-    if (!document.styleScope().resolverIfExists())
-        document.styleScope().didChangeStyleSheetEnvironment();
+    if (!document->styleScope().resolverIfExists())
```
Do we need a `protectedStyleScope()`?

cdumez replied:
1. resolverIfExists() seems like a trivial function.
2. StyleScope is not RefCounted.

We could call `checkedStyleScope()` but given that `resolverIfExists()` seems trivial, not sure it is worth the refcounting churn.

brentfulgham commented on Source/WebCore/page/mac/ServicesOverlayController.mm:
```diff
@@ -648,9 +648,9 @@ static void compactRectsWithGapRects(Vector<LayoutRect>& rects, const Vector<Gap
         });
 
-        m_page.chrome().client().handleSelectionServiceClick(CheckedRef(m_page.focusController())->focusedOrMainFrame().selection(), WTFMove(selectedTelephoneNumbers), windowPoint);
+        m_page->chrome().client().handleSelectionServiceClick(CheckedRef(m_page->focusController())->focusedOrMainFrame().selection(), WTFMove(selectedTelephoneNumbers), windowPoint);
```
Does the `m_page->chrome().client()` need to be protected while the handler runs?

cdumez replied:
This one wasn't apparently reported by the analyzer. I suspect we should protect the page and use `checkedClient()` but I'll need to look at ownership to make sure.

brentfulgham reviewed (approved):
LGTM. I had a couple of questions about life cycle while reading the code, but nothing to prevent you from landing these changes. Possibly ideas for a future iteration.

cdumez committed:
Adopt more smart pointers under WebCore/page/
https://bugs.webkit.org/show_bug.cgi?id=269209

Reviewed by Brent Fulgham.

* Source/WebCore/bindings/js/JSWindowProxy.cpp:
(WebCore::JSWindowProxy::protectedWorld):
* Source/WebCore/bindings/js/JSWindowProxy.h:
* Source/WebCore/dom/Node.cpp:
(WebCore::Node::protectedContainingShadowRoot const):
(WebCore::Node::checkedRenderer const):
* Source/WebCore/dom/Node.h:
* Source/WebCore/html/HTMLFrameOwnerElement.cpp:
(WebCore::HTMLFrameOwnerElement::~HTMLFrameOwnerElement):
(WebCore::HTMLFrameOwnerElement::protectedContentFrame const):
* Source/WebCore/html/HTMLFrameOwnerElement.h:
* Source/WebCore/inspector/agents/page/PageRuntimeAgent.cpp:
(WebCore::PageRuntimeAgent::PageRuntimeAgent):
(WebCore::PageRuntimeAgent::reportExecutionContextCreation):
* Source/WebCore/page/DOMSelection.cpp:
(WebCore::DOMSelection::getComposedRanges):
* Source/WebCore/page/EventHandler.cpp:
(WebCore::EventHandler::updateSelectionForMouseDrag):
(WebCore::EventHandler::hitTestResultAtPoint const):
(WebCore::EventHandler::performDragAndDrop):
(WebCore::EventHandler::dispatchMouseEvent):
(WebCore::EventHandler::sendContextMenuEvent):
(WebCore::EventHandler::dispatchEventToDragSourceElement):
(WebCore::EventHandler::handleDrag):
(WebCore::EventHandler::keyboardScrollRecursively):
* Source/WebCore/page/FocusController.cpp:
(WebCore::FocusController::setFocusedFrame):
(WebCore::FocusController::focusedOrMainFrame const):
(WebCore::FocusController::setFocused):
(WebCore::FocusController::setFocusedInternal):
(WebCore::FocusController::relinquishFocusToChrome):
(WebCore::FocusController::advanceFocusInDocumentOrder):
(WebCore::shouldClearSelectionWhenChangingFocusedElement):
(WebCore::FocusController::setFocusedElement):
(WebCore::FocusController::protectedPage const):
(WebCore::FocusController::setActive):
(WebCore::FocusController::setActiveInternal):
(WebCore::FocusController::setIsVisibleAndActiveInternal):
(WebCore::FocusController::advanceFocusDirectionallyInContainer):
* Source/WebCore/page/FocusController.h:
* Source/WebCore/page/LocalDOMWindow.cpp:
(WebCore::LocalDOMWindow::shouldHaveWebKitNamespaceForWorld):
(WebCore::LocalDOMWindow::webkitNamespace):
(WebCore::LocalDOMWindow::isAllowedToUseDeviceMotion const):
(WebCore::LocalDOMWindow::open):
* Source/WebCore/page/LocalFrame.cpp:
(WebCore::LocalFrame::screenSize const):
* Source/WebCore/page/LocalFrameView.cpp:
(WebCore::LocalFrameView::updateSnapOffsets):
* Source/WebCore/page/LocalFrameViewLayoutContext.cpp:
(WebCore::LocalFrameViewLayoutContext::layout):
(WebCore::LocalFrameViewLayoutContext::performLayout):
(WebCore::LocalFrameViewLayoutContext::runPostLayoutTasks):
(WebCore::LocalFrameViewLayoutContext::scheduleLayout):
(WebCore::LocalFrameViewLayoutContext::scheduleSubtreeLayout):
(WebCore::LocalFrameViewLayoutContext::updateStyleForLayout):
(WebCore::LocalFrameViewLayoutContext::protectedFrame):
(WebCore::LocalFrameViewLayoutContext::view const):
(WebCore::LocalFrameViewLayoutContext::protectedView const):
* Source/WebCore/page/LocalFrameViewLayoutContext.h:
* Source/WebCore/page/Location.cpp:
(WebCore::Location::reload):
* Source/WebCore/page/MouseEventWithHitTestResults.h:
(WebCore::MouseEventWithHitTestResults::protectedTargetNode const):
* Source/WebCore/page/Navigator.cpp:
(WebCore::Navigator::appVersion const):
(WebCore::Navigator::userAgent const):
(WebCore::Navigator::plugins):
(WebCore::Navigator::mimeTypes):
(WebCore::Navigator::cookieEnabled const):
* Source/WebCore/page/Page.cpp:
(WebCore::Page::viewportArguments const):
(WebCore::Page::takeAnyMediaCanStartListener):
(WebCore::Page::setCanStartMedia):
(WebCore::Page::protectedMainFrame const):
(WebCore::Page::forEachDocument const):
* Source/WebCore/page/Page.h:
* Source/WebCore/page/PageConsoleClient.cpp:
(WebCore::PageConsoleClient::addMessage):
(WebCore::PageConsoleClient::messageWithTypeAndLevel):
(WebCore::PageConsoleClient::count):
(WebCore::PageConsoleClient::countReset):
(WebCore::PageConsoleClient::profile):
(WebCore::PageConsoleClient::profileEnd):
(WebCore::PageConsoleClient::takeHeapSnapshot):
(WebCore::PageConsoleClient::time):
(WebCore::PageConsoleClient::timeLog):
(WebCore::PageConsoleClient::timeEnd):
(WebCore::PageConsoleClient::timeStamp):
(WebCore::PageConsoleClient::screenshot):
(WebCore::PageConsoleClient::protectedPage const):
* Source/WebCore/page/PageConsoleClient.h:
* Source/WebCore/page/PageOverlayController.cpp:
(WebCore::PageOverlayController::createRootLayersIfNeeded):
(WebCore::PageOverlayController::installedPageOverlaysChanged):
(WebCore::PageOverlayController::protectedPage const):
(WebCore::PageOverlayController::attachViewOverlayLayers):
(WebCore::PageOverlayController::detachViewOverlayLayers):
(WebCore::PageOverlayController::layerWithDocumentOverlays):
(WebCore::PageOverlayController::layerWithViewOverlays):
(WebCore::PageOverlayController::protectedLayerWithViewOverlays):
(WebCore::PageOverlayController::installPageOverlay):
(WebCore::PageOverlayController::updateForceSynchronousScrollLayerPositionUpdates):
(WebCore::PageOverlayController::didChangeViewExposedRect):
(WebCore::PageOverlayController::updateSettingsForLayer):
(WebCore::PageOverlayController::deviceScaleFactor const):
(WebCore::PageOverlayController::notifyFlushRequired):
(WebCore::PageOverlayController::didChangeOverlayFrame):
(WebCore::PageOverlayController::didChangeOverlayBackgroundColor):
(WebCore::PageOverlayController::tiledBackingUsageChanged):
* Source/WebCore/page/PageOverlayController.h:
* Source/WebCore/page/PerformanceMonitor.cpp:
(WebCore::PerformanceMonitor::didFinishLoad):
(WebCore::PerformanceMonitor::activityStateChanged):
(WebCore::PerformanceMonitor::measurePostLoadCPUUsage):
(WebCore::PerformanceMonitor::measurePostLoadMemoryUsage):
(WebCore::PerformanceMonitor::measurePostBackgroundingMemoryUsage):
(WebCore::PerformanceMonitor::measurePostBackgroundingCPUUsage):
(WebCore::PerformanceMonitor::measurePerActivityStateCPUUsage):
(WebCore::PerformanceMonitor::measureCPUUsageInActivityState):
* Source/WebCore/page/PerformanceMonitor.h:
* Source/WebCore/page/PointerLockController.cpp:
(WebCore::PointerLockController::didAcquirePointerLock):
(WebCore::PointerLockController::didNotAcquirePointerLock):
(WebCore::PointerLockController::didLosePointerLock):
(WebCore::PointerLockController::enqueueEvent):
* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::needsAutoplayPlayPauseEvents const):
(WebCore::Quirks::needsPerDocumentAutoplayBehavior const):
(WebCore::Quirks::shouldSynthesizeTouchEvents const):
(WebCore::Quirks::shouldBypassBackForwardCache const):
(WebCore::Quirks::shouldAvoidPastingImagesAsWebContent const):
(WebCore::Quirks::requestStorageAccessAndHandleClick const):
(WebCore::Quirks::protectedDocument const):
(WebCore::Quirks::triggerOptionalStorageAccessIframeQuirk const):
(WebCore::Quirks::triggerOptionalStorageAccessQuirk const):
(WebCore::Quirks::needsDisableDOMPasteAccessQuirk const):
* Source/WebCore/page/Quirks.h:
* Source/WebCore/page/ResourceUsageOverlay.cpp:
(WebCore::ResourceUsageOverlay::~ResourceUsageOverlay):
(WebCore::ResourceUsageOverlay::initialize):
(WebCore::ResourceUsageOverlay::mouseEvent):
* Source/WebCore/page/Screen.cpp:
(WebCore::fingerprintingProtectionsEnabled):
(WebCore::Screen::height const):
(WebCore::Screen::width const):
(WebCore::Screen::colorDepth const):
(WebCore::Screen::availLeft const):
(WebCore::Screen::availTop const):
(WebCore::Screen::availHeight const):
(WebCore::Screen::availWidth const):
(WebCore::Screen::orientation):
* Source/WebCore/page/SpatialNavigation.cpp:
(WebCore::FocusCandidate::FocusCandidate):
(WebCore::canScrollInDirection):
(WebCore::nodeRectInAbsoluteCoordinates):
(WebCore::virtualRectForAreaElementAndDirection):
(WebCore::frameOwnerElement):
* Source/WebCore/page/SpatialNavigation.h:
* Source/WebCore/page/WebKitNamespace.cpp:
(WebCore::WebKitNamespace::WebKitNamespace):
* Source/WebCore/page/cocoa/WebTextIndicatorLayer.mm:
(-[WebTextIndicatorLayer present]):
* Source/WebCore/page/mac/EventHandlerMac.mm:
(WebCore::EventHandler::passWidgetMouseDownEventToWidget):
* Source/WebCore/page/mac/ServicesOverlayController.h:
* Source/WebCore/page/mac/ServicesOverlayController.mm:
(WebCore::ServicesOverlayController::invalidateHighlightsOfType):
(WebCore::ServicesOverlayController::buildPotentialHighlightsIfNeeded):
(WebCore::ServicesOverlayController::remainingTimeUntilHighlightShouldBeShown const):
(WebCore::ServicesOverlayController::buildPhoneNumberHighlights):
(WebCore::ServicesOverlayController::buildSelectionHighlight):
(WebCore::ServicesOverlayController::hasRelevantSelectionServices):
(WebCore::ServicesOverlayController::createOverlayIfNeeded):
(WebCore::ServicesOverlayController::telephoneNumberRangesForFocusedFrame):
(WebCore::ServicesOverlayController::findTelephoneNumberHighlightContainingSelectionHighlight):
(WebCore::ServicesOverlayController::mouseEvent):
(WebCore::ServicesOverlayController::handleClick):
* Source/WebCore/rendering/HitTestResult.cpp:
(WebCore::HitTestResult::protectedTargetNode const):
(WebCore::HitTestResult::protectedTargetElement const):
* Source/WebCore/rendering/HitTestResult.h:
* Source/WebCore/rendering/RenderWidget.h:
(WebCore::RenderWidget::protectedWidget const):

Canonical link: https://commits.webkit.org/274488@main
